### ISSUE TITLE
Aarch64 support?

### ISSUE THREAD

**Issue by vanfanel** [M-HT/SR#28]
Hi!

I would like to run Albion on the Raspberry Pi 400 using my minimal aarch64 system. Are there support plans for generating aarch64 binaries?

Thanks!

**Comment by M-HT** [comment]
I'm planning to look into it, but there's no timeline.

Generating code which uses 64-bit pointers, when the original code uses 32-bit pointers is not easy.

If ILP32 for AArch64 was usable (which it isn't), then generating binaries for it wouldn't be very difficult.

**Comment by FyiurAmron** [comment]
@vanfanel did you try running it using e.g. https://wiki.debian.org/Arm64ilp32Port or something similar? I understand that it requires custom kernel modification, but I'm still curious if it would be possible to run current binaries this way. (also, FWIW, weren't ILP32 [already possible on AArch64](https://documentation-service.arm.com/static/5f033160cafe527e86f5b8c1?token=)? I guess the real-life usability @M-HT talks about is the problem here; I haven't tried it myself ever, so I'm only throwing stuff I've heard about around)

**Comment by M-HT** [comment]
@FyiurAmron you can't run 32-bit arm binaries on ILP32 for AArch64, you need to compile programs for this architecture. You can't compile Albion for it at the moment, but it wouldn't be difficult to add support for it - that means adding llasm version. Septerra Core has a llasm version, so it should be possible to compile it for ILP32 for AArch64.

As for ILP32 for AArch64 itself, it's my understanding that it was worked on a few years ago, but it's not usable on any distribution. If I'm wrong about it, anyone is free to correct me.

**Comment by M-HT** [comment]
Nothing new about aarch64 binaries, but I want to point out that as long as the CPU supports 32-bit mode (which Raspberry Pi 400 does) you can create a 32-bit chroot and run 32-bin binaries from there.

I also tried compiling a kernel with ILP32 enabled, but my Raspberry Pi didn't boot, so no progress in this direction either.

**Comment by vanfanel** [comment]
@M-HT The idea about a 32-bit chroot sounds very good indeed...
I guess a 32-bit chroot rootfs can live in a directory inside the 64-bit (aarch64) system, right?
Then, after booting in aarch64 mode as I normally do, I would have to chroot into the 32-bit system and run the games... Am I right on this? If that's the case, it seems like a very nice experiment to do at home! :)

**Comment by M-HT** [comment]
@vanfanel Yes, that's how it works. When I tested it, I took some inspiration from [this tutorial](https://forum.armbian.com/topic/16584-install-box86-on-arm64/) - it's not the best way to create chroot, so I didn't follow it exactly.

**Comment by neofeo** [comment]
Multiarch is another option. Did you tested to compile anyone @vanfanel ohh, you are the sdl2 guy.

**Comment by M-HT** [comment]
I added support for generating 64-bit (x64, arm64) Linux version of Albion executable.
It's still experimental, but should be usable.

**Comment by M-HT** [comment]
I added x64 and arm64 versions of Albion and Septerra Core to the Releases.

**Comment by ralph303** [comment]
Would it be possible to compile a version for MacOS, now that there is support for the ARM architecture?
That would be really great, as it's the only game for which I still need to launch my "Windows 11 ARM @ VMWare Fusion" :)

**Comment by M-HT** [comment]
> Would it be possible to compile a version for MacOS, now that there is support for the ARM architecture?

I don't work with MacOS, so I can't answer the question at this moment.

**Comment by M-HT** [comment]
I created new releases for all games (except Battle Isle 3) which include versions for MacOS (X64 and ARM).
I only have a Mac with x64 CPU, so I can't test the ARM version, but it hopefully works like the X64 version.

**Comment by ralph303** [comment]
Thanks a lot for supplying that version!
It seems that some part is still compiled in x64 code, because it wants to install Rosetta at startup.

**Comment by M-HT** [comment]
I fixed the package, please redownload it and test if it's ok.

**Comment by ralph303** [comment]
It works and runs like a charm, thank you!
How can I reference to a soundfont which is in a directory containing a blank? I tried backslash for escaping and double quotes, but that didn't work out.

**Comment by M-HT** [comment]
> How can I reference to a soundfont which is in a directory containing a blank? I tried backslash for escaping and double quotes, but that didn't work out.

Just entering it normally (without backslash or double quotes) should work. Something like this:
```
Audio_SoundFont_Path=/Users/name/path with spaces/soundfont.sf2
```

**Comment by ralph303** [comment]
That was too easy, so I didn't even bother to try it like this :)
Thanks!
